### ISSUE TITLE
Scroll handle is not positionned properly when scrolling the recycling view

### ISSUE THREAD

**Issue by bperel** [quiph/RecyclerView-FastScroller#45]
Hello and thank you for this project.

I have noticed an issue when the list contains a small number of elements.
It can be reproduced by only keeping the first 20 or so items in the `countries.json` file of the sample app.

When scrolling using the scroll handle everything is going fine : 
<img src="https://user-images.githubusercontent.com/1484692/103479961-803e4400-4dd1-11eb-98cf-4637e5909bad.jpg" width=200 />

but when dragging down the list _itself_ to the end, the scroll handle doesn't go until the bottom, even though I have reached the last element in the list : 
<img src="https://user-images.githubusercontent.com/1484692/103479971-9fd56c80-4dd1-11eb-9809-2e1e4fa1dee4.jpg" width=200 />

Is it related to the TODO item mentioned in the README ( `Make handle size flexible to item count in adapter`) or is it a bug?

**Comment by shahsurajk** [comment]
Hmm, this surely looks like a bug and is not related to the TODO mentioned. Thanks for letting us know!

**Comment by AndroidDeveloperLB** [comment]
Is this issue the same as I show here, after modifying the sample a bit:

[KK5iETTupT.zip](https://github.com/quiph/RecyclerView-FastScroller/files/5823462/KK5iETTupT.zip)

[RecyclerView-FastScroller.zip](https://github.com/quiph/RecyclerView-FastScroller/files/5823463/RecyclerView-FastScroller.zip)

?

BasicAdapter - changed number of items to 10.
recycler_view_list_item_center.xml - changed item height to 100dp

Maybe this library could help fix this:
https://github.com/zhanghai/AndroidFastScroll
?

**Comment by bperel** [comment]
@AndroidDeveloperLB Yes your video is the issue that I have

**Comment by AndroidDeveloperLB** [comment]
@bperel Thank you. Do you know of any workaround of this?

**Comment by bperel** [comment]
No, I didn't find a workaround: my project can't use the library until this bug is fixed since it prevents the user from scrolling through the content properly

**Comment by AndroidDeveloperLB** [comment]
@bperel OK thank you.

**Comment by BenjiLee** [comment]
Since this project hasn't been updated in a year and I haven't extensively tested this solution, I'm leaving this here as a comment instead of submitting a PR in case anyone else has come across this issue.

There seems to be an issue with the math `onScrolled`

Ended up replacing https://github.com/quiph/RecyclerView-FastScroller/blob/v1.0.0/recyclerviewfastscroller/src/main/java/com/qtalk/recyclerviewfastscroller/RecyclerViewFastScroller.kt#L906-L909 with

```
val availableTrackLength = extent.toFloat() - handleLength
val scrollPositionPercent = (offset.toFloat() / (range - extent))
val finalOffset = availableTrackLength * scrollPositionPercent

moveHandle(finalOffset)
```

**Comment by T1mmyK0** [comment]
I have the same issue.
